postgres/lib/PostgresService.js

```diff
@@ -178,26 +177,31 @@ class PostgresService extends SQLService {
   }
 
   static CQN2SQL = class CQN2Postgres extends SQLService.CQN2SQL {
-    orderBy(orderBy, localized) {
+    _orderBy(orderBy, localized, locale) {
       return orderBy.map(
         localized
           ? c =>
               this.expr(c) +
-              (c.element?.[this.class._localized] ? ` COLLATE "${this.context.locale}"` : '') +
+              (c.element?.[this.class._localized] ? ` COLLATE "${locale}"` : '') +
               (c.sort === 'desc' || c.sort === -1 ? ' DESC' : ' ASC')
           : c => this.expr(c) + (c.sort === 'desc' || c.sort === -1 ? ' DESC' : ' ASC'),
       )
     }
 
+    orderBy(orderBy, localized) {
+      // best-effort locale expansion (example: 'de' -> 'de_DE')
+      const locale =
+        this.context.locale === 'en'
```

Comment:
If the translation from `de` to `de_DE` is something we want to do it would be better to create a map based upon the available collations in the database.

After determining that the database does not include `icu` it would be possible to do a query like:
```sql
SELECT collname FROM pg_collation WHERE collprovider='c'
```

Which returns all the available `c` collations and it would be possible to use the first two characters as `key` and then assign the collation name as value. The main issue being that the collations don't seem to have a priority defined https://www.postgresql.org/docs/current/catalog-pg-collation.html. Which means that it might use `de_AT` when translating `de`.

Additionally it is problematic to provide `c` support without a way of testing it. Locally I only have 3 `c` provided collate options.

Additionally with the current collation check it is still possible that `icu` and `c` are both missing from the system and the exact same error would comeback it would only say `en_US`/`de_DE` instead of `en`/`de`.

Reply:
yeah, i wanted to save another round trip. but i guess one or two additional selects during startup don't hurt too much...

actually, i think the variant shouldn't matter as we use only for ordering, right? the austrians probably sort the same as the germans and the australians the same as the british, no?

in case both are missing, i'd vote for default rather than an error.